### PR TITLE
fix xray trace/span id transfer

### PR DESCRIPTION
**Pull request by wyTrivail**
**Description:**  Fix a bug in XRay Receiver 

Today AWSXRayReceiver directly use the encoded hex value, which are supposed to be decoded, to transfer XRay trace/span id to OTLP format.

This bug leads to an issue that the trace id on XRay Console is different with the one emitted from XRay SDK. (we're lucky because XRayExporter generate a new trace id when the incoming trace id is invalid, for this we have another issue to discuss: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1255)

**Testing:** 

* added and modified Unit Tests
* manually test to verify we can see the same trace id on XRay

### CODE REVIEW

File: receiver/awsxrayreceiver/internal/translator/translator.go [hunk 1]
```diff
@@ -189,3 +226,27 @@ func totalSegmentsCount(seg awsxray.Segment) int {
...
+	if len(*traceID) < 35 {
+		return nil, errors.New("traceID length is wrong")
+	}
+	traceIDtoBeDecoded := (*traceID)[2:10] + (*traceID)[11:]
+	return hex.DecodeString(traceIDtoBeDecoded)
+}
+
+// decodeXRaySpanID decodes the spanid from xraysdk
+func decodeXRaySpanID(spanID *string) ([]byte, error) {
+	if spanID == nil {
```
Comment (anuraaga) [inline receiver/awsxrayreceiver/internal/translator/translator.go:248]: Can you check the length here too?

**Review by anuraaga — APPROVED** [review]
Just small nit, thanks!

File: receiver/awsxrayreceiver/internal/translator/translator.go (hunk 1 above)
Comment (wyTrivail) [reply receiver/awsxrayreceiver/internal/translator/translator.go:248]: @anuraaga  thx! 

is the XRay Span Id length always 16?  
i'm a little concerned that adding length limit here could potentially break span id if XRay changes the length.

Comment (anuraaga) [reply receiver/awsxrayreceiver/internal/translator/translator.go:248]: Yeah it is - OTel span ID, which we are converting to here, always has the length so X-Ray can't change it really. If something comes up in the distant future, we'll revisit then

Comment (wyTrivail) [reply receiver/awsxrayreceiver/internal/translator/translator.go:248]: okay, let me add that limit then!

Comment (bogdandrutu) [reply receiver/awsxrayreceiver/internal/translator/translator.go:248]: Did you fix this comment? Does not seem to be a commit after.

Comment (wyTrivail) [reply receiver/awsxrayreceiver/internal/translator/translator.go:248]: Will do that today

Comment (wyTrivail) [reply receiver/awsxrayreceiver/internal/translator/translator.go:248]: done

**Pull request merged**